kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemDataset.java

```diff
@@ -463,6 +469,13 @@ public long getLastModified() {
         throw new DatasetIOException("Cannot find last modified time of of " + dir, e);
       }
     }
+
+    // use the ready time if it is more recent
+    long viewModified = unbounded.getLastModified();
```

Comment:
This delegation will recursively check the file system, so the logic above can be removed. I think this can just delegate to `unbounded.getLastModified()`